### PR TITLE
Implement the Array Grouping tests

### PR DESCRIPTION
**Pull request by afmenez**
I am not sure about the 'significance' field.

### CODE REVIEW

File: data-es2016plus.js [hunk 1]
```diff
@@ -7361,6 +7361,61 @@ exports.tests = [
...
+    subtests: [
+      {
+        name: 'Object.groupBy()',
+        mdn: 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy',
+        exec: function () {/*
+          const array = [1, 2, 3, 4];
+          const obj = Object.groupBy(array, (num, index) => {
+            return num % 2 === 0 ? 'even': 'odd';
+          });
+          return obj['even'][0] == 2 && obj['odd'][0] == 1;
```
Comment (ljharb) [inline data-es2016plus.js]:
```suggestion
          return !('toString' in obj) && obj.even[0] == 2 && obj.odd[0] == 1;
```
to ensure it's got a null prototype

File: data-es2016plus.js [hunk 2]
```diff
@@ -7361,6 +7361,61 @@ exports.tests = [
...
+        name: 'Map.groupBy()',
+        mdn: 'https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/groupBy',
+        exec: function () {/*
+          const array = [1, 2, 3, 4];
+          const odd  = { odd: true };
+          const even = { even: true };
+          const map = Map.groupBy(array, (num, index) => {
+            return num % 2 === 0 ? even: odd;
+          });
+          return map.get(even)[0]  == 2 && map.get(odd)[0]  == 1;
```
Comment (ljharb) [inline data-es2016plus.js]:
```suggestion
          return map instanceof Map && map.get(even)[0] === 2 && map.get(odd)[0] === 1;
```

**Review by ljharb — CHANGES_REQUESTED** [review]

File: data-es2016plus.js (hunk 1 above)
Comment (afmenez) [reply data-es2016plus.js]: Done.

File: data-es2016plus.js (hunk 2 above)
Comment (afmenez) [reply data-es2016plus.js]: Done.

**Comment by afmenez** [comment]
I had to rebase to commit the requested changes, but now the comments are gone.

**Review by chicoxyzzy — APPROVED** [review]

**Review by ljharb — APPROVED** [review]

**Pull request merged**